src/nvim/lua/stdlib.c

```diff
@@ -305,7 +305,11 @@ int nlua_regex(lua_State *lstate)
     api_clear_error(&err);
     return lua_error(lstate);
   }
-  assert(prog);
+  else if (prog == NULL) {
+    nlua_push_errstr(lstate, "couldn't parse regex");
+    api_clear_error(&err);
```

Comment:
`api_clear_error(&err)` returns immdiately if `ERROR_SET(&err)` is `false`, so this line is not needed.
```suggestion
```